### PR TITLE
Metris complete refactoring

### PR DESCRIPTION
**Pull request by maksd**
Changes proposed in this pull request:

- adding unit tests
- adding new metrics
- adding caching
- adding grafana dashboard to chart
- adding prometheus rules to chart

### CODE REVIEW

**Review by ebensom — DISMISSED** [review]

File: components/metris/README.md [hunk 1]
```diff
@@ -8,21 +8,21 @@ Metris is a metering component that collects data and sends them to EDP.
...
+| `--edp-workers` | **EDP_WORKERS** | Number of workers to send metrics. | `5` |
+| `--edp-event-retry` | **EDP_RETRY** | Number of retries for sending event. | `5` |
+| `--provider-poll-interval` | **PROVIDER_POLLINTERVAL** | Interval at which metrics are fetch. | `1m` |
+| `--provider-workers` | **PROVIDER_WORKERS** | Number of workers to fetch metrics. | `10` |
+| `--provider-buffer` | **PROVIDER_BUFFER** | Number of cluster that the buffer can have. | `100` |
+| `--listen-addr` | **METRIS_LISTEN_ADDRESS** | Address and port the metrics and health HTTP endpoints will bind to. | None |
+| `--debug-port` | **METRIS_DEBUG_PORT** | Port the debug HTTP endpoint will bind to. Always listen on localhost. | None |
+| `--config-file` | None | Location of the config file. | None |
+| `--kubeconfig` | **METRIS_KUBECONFIG** | Path to the Gardener kubeconfig file. | None |
+| `--log-level` | **METRIS_LOGLEVEL** | Logging level. (debug,info,warn,error) | `info` |
```
Comment (alexandra-simeonova) [inline components/metris/README.md:28]: Keeping the entries consistent with earlier ones, including formatting and periods at the end of phrases.
```suggestion
| `--edp-buffer` | **EDP_BUFFER** | Number of events that the buffer can have | `100` |
| `--edp-workers` | **EDP_WORKERS** | Number of workers to send metrics | `5` |
| `--edp-event-retry` | **EDP_RETRY** | Number of retries for sending an event | `5` |
| `--provider-poll-interval` | **PROVIDER_POLLINTERVAL** | Interval at which metrics are fetched | `1m` |
| `--provider-workers` | **PROVIDER_WORKERS** | Number of workers to fetch metrics | `10` |
| `--provider-buffer` | **PROVIDER_BUFFER** | Number of clusters that the buffer can have | `100` |
| `--listen-addr` | **METRIS_LISTEN_ADDRESS** | Address and port the metrics and health HTTP endpoints will bind to | None |
| `--debug-port` | **METRIS_DEBUG_PORT** | Port the debug HTTP endpoint will bind to (always listen on localhost) | None |
| `--config-file` | None | Location of the `config` file | None |
| `--kubeconfig` | **METRIS_KUBECONFIG** | Path to the Gardener `kubeconfig` file | None |
| `--log-level` | **METRIS_LOGLEVEL** | Logging level (`debug`,`info`,`warn`,`error`) | `info` |
```

**Review by alexandra-simeonova — CHANGES_REQUESTED** [review]
Technical writer review

**Review by alexandra-simeonova — APPROVED** [review]

**Pull request merged**